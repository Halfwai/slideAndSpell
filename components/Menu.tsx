import { Text, View, Image, StyleSheet } from "react-native";
import MenuButton from "@/components/MenuButton";
import React, {useState} from "react";

import { useRouter, RelativePathString } from 'expo-router';

export default function Menu(){
    const [exitMenu, setExitMenu] = useState(false);

    const router = useRouter();

    const switchRoute = (route : RelativePathString) => {
        setExitMenu(true),
        setTimeout(() => {
            router.push(route);
            setExitMenu(false);
        }, 700);        
    }
    return (
        <View
            style={styles.container}
        >
            <Image 
                source={require("@/assets/images/logo.png")}
                style={{ flex: 1, width: "90%" }}
                resizeMode="contain"
            />
            <View style={{
                flex: 2,
                flexDirection: 'column',
                width: "100%",
            }}>
                <MenuButton 
                    text="Puzzle of the Day"
                    onPress={() => (
                        console.log("Puzzle of the Day"),
                        switchRoute('/puzzleOfTheDay' as RelativePathString)
                    )}
                    delay={0}
                    exitMenu={exitMenu}
                />
                <MenuButton 
                    text="Endless Mode"
                    onPress={() => (
                        console.log("Endless Mode"),
                        switchRoute('/endlessGame' as RelativePathString)
                    )}
                    delay={100}
                    exitMenu={exitMenu}
                />
                <MenuButton 
                    text="Stats"
                    onPress={() => (
                        console.log("Stats")
                    )}
                    delay={200}
                    exitMenu={exitMenu}
                />
                <MenuButton 
                    text="Leaderboards"
                    onPress={() => (
                        console.log("Leaderboards")
                    )}
                    delay={300}
                    exitMenu={exitMenu}
                />
                <MenuButton 
                    text="Settings"
                    onPress={() => (
                        console.log("Settings")
                    )}
                    delay={400}
                    exitMenu={exitMenu}
                />
                <View style={{ flexDirection: 'row', flex: 1, width: "100%", justifyContent: "space-evenly", alignItems: "center" }}>
                <View style={{ width: "40%", height: "100%" }} >
                    <MenuButton 
                        text="Log Out"
                        onPress={() => (
                            console.log("LogOut")
                        )}
                        delay={500}
                        exitMenu={exitMenu}
                    />
                </View>
                <View style={{ width: "40%", height: "100%" }} >
                    <MenuButton 
                        text="Exit"
                        onPress={() => (
                            console.log("Exit")
                        )}
                        delay={500}
                        exitMenu={exitMenu}
                    />
                </View>               
            </View>
            </View>

        </View>
    );
    
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
        backgroundColor: "#CBE6F7",
    },
});