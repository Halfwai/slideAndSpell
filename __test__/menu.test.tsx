import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

import Menu from '@/app/menu';

beforeEach(() => {
    jest.useFakeTimers();
});

afterEach(() => {
    jest.clearAllMocks();
    jest.useRealTimers();
});

jest.mock('@/lib/supabase', () => ({
    supabase: {
        auth: {
            signOut: jest.fn(),
        },
    },
}));

it(`Menu Component renders correctly`, () => {
    const { toJSON } = render(
        <NavigationContainer>
            <Menu/>
        </NavigationContainer>
    );
    expect(toJSON()).toMatchSnapshot();
});