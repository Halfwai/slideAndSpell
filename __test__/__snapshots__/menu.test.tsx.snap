// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#CBE6F7",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../assets/images/logo.png",
      }
    }
    style={
      {
        "height": 400.2,
        "width": "90%",
      }
    }
  />
  <View
    style={
      {
        "flex": 2,
        "flexDirection": "column",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "translateX": 750,
              },
            ],
            "width": "80%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#8BF0BC",
              "borderRadius": 10,
              "borderWidth": 1,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 10,
            }
          }
          testID="menuButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                {
                  "color": "black",
                  "fontSize": 51,
                  "textAlign": "center",
                },
              ]
            }
          >
            Puzzle of the Day
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "translateX": 750,
              },
            ],
            "width": "80%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#8BF0BC",
              "borderRadius": 10,
              "borderWidth": 1,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 10,
            }
          }
          testID="menuButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                {
                  "color": "black",
                  "fontSize": 51,
                  "textAlign": "center",
                },
              ]
            }
          >
            Endless Mode
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "translateX": 750,
              },
            ],
            "width": "80%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#8BF0BC",
              "borderRadius": 10,
              "borderWidth": 1,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 10,
            }
          }
          testID="menuButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                {
                  "color": "black",
                  "fontSize": 51,
                  "textAlign": "center",
                },
              ]
            }
          >
            Stats
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "translateX": 750,
              },
            ],
            "width": "80%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#8BF0BC",
              "borderRadius": 10,
              "borderWidth": 1,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 10,
            }
          }
          testID="menuButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                {
                  "color": "black",
                  "fontSize": 51,
                  "textAlign": "center",
                },
              ]
            }
          >
            Leaderboards
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "translateX": 750,
              },
            ],
            "width": "80%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#8BF0BC",
              "borderRadius": 10,
              "borderWidth": 1,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 10,
            }
          }
          testID="menuButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                {
                  "color": "black",
                  "fontSize": 51,
                  "textAlign": "center",
                },
              ]
            }
          >
            Settings
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "height": "100%",
            "paddingBottom": "10%",
            "width": "40%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "marginVertical": 5,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "translateX": 750,
                  },
                ],
                "width": "80%",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderColor": "#8BF0BC",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 1,
                  "padding": 10,
                }
              }
              testID="menuButton"
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Quicksand",
                    },
                    {
                      "color": "black",
                      "fontSize": 51,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Log Out
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "height": "100%",
            "paddingBottom": "10%",
            "width": "40%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "marginVertical": 5,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "translateX": 750,
                  },
                ],
                "width": "80%",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderColor": "#8BF0BC",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 1,
                  "padding": 10,
                }
              }
              testID="menuButton"
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Quicksand",
                    },
                    {
                      "color": "black",
                      "fontSize": 51,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Exit
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
